server/routes/user/usersettings.ts

```diff
@@ -99,11 +100,29 @@ userSettingsRoutes.post<
       });
     }
 
-    user.username = req.body.username;
     const oldEmail = user.email;
+    const oldUsername = user.username;
+    user.username = req.body.username;
     if (user.jellyfinUsername) {
       user.email = req.body.email || user.jellyfinUsername || user.email;
     }
+    // Edge case for local users, because they have no Jellyfin username to fall back on
+    // if the email is not provided
+    if (user.userType === UserType.LOCAL) {
+      if (req.body.email) {
+        user.email = req.body.email;
+        if (
+          !user.username &&
+          user.email !== oldEmail &&
+          !oldEmail.includes('@')
+        ) {
+          user.username = oldEmail;
+        }
+      } else if (req.body.username) {
+        user.email = oldUsername || user.email;
```

Comment:
Wouldn't it better to use nullish Coalescing here to avoid assigning falsify values like an empty string when user.email should never be empty?

```suggestion
        user.email = oldUsername ?? user.email;
```